### PR TITLE
Add the live streaming message endpoint for a particular room

### PR DESCRIPTION
**Pull request by laserlemon**
This doesn't test the XML endpoint but JSON is recommended anyway.

### CODE REVIEW

File: spec/factories/messages.rb [hunk 1]
```diff
@@ -12,4 +12,7 @@
       private true
     end
   end
+
+  factory :text_message,  parent: :message, class: TextMessage
+  factory :paste_message, parent: :message, class: PasteMessage
```
Comment (emilford) [inline spec/factories/messages.rb]: This way might be your preference, but you could define these inside the `factory :message` block to the get the inheritance without the `parent` options.

File: spec/requests/streaming_spec.rb [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+              "created_at" => message.created_at,
+              "id" => message.id,
+              "room_id" => message.room_id,
+              "type" => message.type,
+              "user_id" => message.user_id
+            )
+
+            expect {
+              create(:text_message)
+              sleep 0.2 # Wait for stream
```
Comment (emilford) [inline spec/requests/streaming_spec.rb:53]: Specifically need a longer wait here?  Anything we can do to get away from `sleep` or are these passing reliably enough as is?

File: spec/factories/messages.rb (hunk 1 above)
Comment (laserlemon) [inline spec/factories/messages.rb]: Didn't know that. Awesome!

File: spec/requests/streaming_spec.rb (hunk 2 above)
Comment (laserlemon) [inline spec/requests/streaming_spec.rb:53]: They seem to be reliable. Not much you can do about avoiding some kind of wait when you have disjointed session interacting with each other.

I added the longer wait for the assertions that a message _doesn't_ come through, for a little added confidence.

**Comment by laserlemon** [comment]
@emilford Updated.

**Pull request merged**